Source/VCGeneration/Checker.cs

```diff
@@ -153,8 +153,13 @@ private void SetRlimit(uint rlimit)
     /// </summary>
     private void Setup(Program prog, ProverContext ctx, Split split = null)
     {
-      SolverOptions.RandomSeed = split?.RandomSeed ?? Options.RandomSeed;
+      if (Options.VcsStabilityIterations < 2 || SolverOptions.RandomSeed == null) {
```

Comment:
Do we need the `Options.VcsStabilityIterations < 2` ?

Would `SolverOptions.RandomSeed ??= split?.RandomSeed ?? Options.RandomSeed` not work?

Consider not checking `Options.VcsStabilityIterations`:

```c#
private void Setup(Program prog, ProverContext ctx, Split split = null)
{
  if (SolverOptions.RandomSeed == null) {
    SolverOptions.RandomSeed = split?.RandomSeed ?? Options.RandomSeed
  } else {
    SolverOptions.RandomSeed = new Random(SolverOptions.RandomSeed.Value).Next();
  }
  var random = SolverOptions.RandomSeed == null ? null : new Random(SolverOptions.RandomSeed.Value);

  Program = prog;
  ..
}
```

Reply:
The first time I wrote it, I did exactly that, but it makes it impossible to use different random seeds for different methods using the `{:random_seed N}` attribute (which we fortunately have a test to check!).